front/src/App.js

```diff
@@ -37,8 +37,9 @@ function App() {
   const [cursorPosition, setCursorPosition] = useState()
   const [opponentCursorPositions, setOpponentCursorPositions] = useState({})
   const [wrongChars, setWrongChars] = useState()
-  const [messages, setMessages] = useState([])
+  const [dataMessages, setDataMessages] = useState([]) //An array contains objects with structure {sender: sender, content: content, playerID: playerID}
   const [socket, setSocket] = useState()
+  const [playerID, setPlayerID] = useState() // Player ID will store here to use when send a message
```

Comment:
I don't find it necessary to store the playerID client-side. I think just emitting the `socket.id` along with the messages is better.

With the current solution the client can change the playerID to masquerade as another player too.

Reply:
Hmm, so if I guest correctly, player does not need to send back the playerID when emit the chat message on client ? Socket.io will remain the socket.id for client's connection ?

Reply:
From my understanding that is how it works.
https://socket.io/docs/server-api/#socket-id